treatment-service/config/config.go

```diff
@@ -24,16 +26,17 @@ type Config struct {
 	Port       int      `json:"port" default:"8080" validate:"required"`
 	ProjectIds []string `json:"project_ids" default:""`
 
-	AssignedTreatmentLogger AssignedTreatmentLoggerConfig       `json:"assigned_treatment_logger"`
-	DebugConfig             DebugConfig                         `json:"debug_config" validate:"required,dive"`
-	NewRelicConfig          newrelic.Config                     `json:"new_relic_config"`
-	SentryConfig            sentry.Config                       `json:"sentry_config"`
-	DeploymentConfig        DeploymentConfig                    `json:"deployment_config" validate:"required,dive"`
-	MessageQueueConfig      common_mq_config.MessageQueueConfig `json:"message_queue_config" validate:"required,dive"`
-	ManagementService       ManagementServiceConfig             `json:"management_service" validate:"required,dive"`
-	MonitoringConfig        Monitoring                          `json:"monitoring_config"`
-	SwaggerConfig           SwaggerConfig                       `json:"swagger_config" validate:"required,dive"`
-	SegmenterConfig         map[string]interface{}              `json:"segmenter_config"`
+	AssignedTreatmentLogger AssignedTreatmentLoggerConfig        `json:"assigned_treatment_logger"`
+	DebugConfig             DebugConfig                          `json:"debug_config" validate:"required,dive"`
+	NewRelicConfig          newrelic.Config                      `json:"new_relic_config"`
+	SentryConfig            sentry.Config                        `json:"sentry_config"`
+	DeploymentConfig        DeploymentConfig                     `json:"deployment_config" validate:"required,dive"`
+	MessageQueueConfig      *common_mq_config.MessageQueueConfig `json:"message_queue_config"`
+	ManagementService       ManagementServiceConfig              `json:"management_service" validate:"required,dive"`
+	MonitoringConfig        Monitoring                           `json:"monitoring_config"`
+	SwaggerConfig           SwaggerConfig                        `json:"swagger_config" validate:"required,dive"`
+	SegmenterConfig         map[string]interface{}               `json:"segmenter_config"`
+	PollerConfig            *PollerConfig                        `json:"poller_config"`
```

Comment:
Can we just use Non Pointer? so that you don't need to check both nil check and enabled check and simply check for enabled?

```
if PoolerConfig != nil && PoolerConfig.Enabled {
```
Instead, can just do
```
if PoolerConfig.Enabled {

```

Reply:
Yes, I have changed it to non pointer. Thanks for mentioning that.